File: DraftingDemo/Visual Basic/DraftingDemo/TangentsForm.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms


	Partial Public Class TangentsForm
		Inherits Form

		Public Sub New()
			InitializeComponent()
			Me.lineRadioButton.Checked = True

		End Sub
        
		Private Sub selectButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles selectButton.Click
			Me.Close()
			Me.DialogResult = System.Windows.Forms.DialogResult.OK

		End Sub

		Private Sub radioButton1_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles lineRadioButton.CheckedChanged
			CircleTangents = circlesRadioButton.Checked
			LineTangents = lineRadioButton.Checked
			radiusLabel.Enabled = circlesRadioButton.Checked
			radiusTextBox.Enabled=circlesRadioButton.Checked
			optionsGruopBox.Enabled=circlesRadioButton.Checked

		End Sub

		Private Sub circlesRadioButton_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles circlesRadioButton.CheckedChanged
			CircleTangents = circlesRadioButton.Checked
			LineTangents = lineRadioButton.Checked
			radiusLabel.Enabled = circlesRadioButton.Checked
			radiusTextBox.Enabled=circlesRadioButton.Checked
			optionsGruopBox.Enabled=circlesRadioButton.Checked
		End Sub

		Private Sub textBox1_TextChanged(ByVal sender As Object, ByVal e As EventArgs) Handles radiusTextBox.TextChanged
			Dim val As Double = Nothing
			If Double.TryParse(radiusTextBox.Text, val) Then
			    TangentRadius = val
			End If

		End Sub
        Public TangentRadius As Double=10
		

		Public Property LineTangents() As Boolean

		Public Property CircleTangents() As Boolean

		Public Property TrimTangents() As Boolean

		Public Property FlipTangents() As Boolean

		Private Sub checkBox2_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles trimCheckBox.CheckedChanged
			TrimTangents = trimCheckBox.Checked
		End Sub

		Private Sub flipCheckBox_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles flipCheckBox.CheckedChanged
			FlipTangents = flipCheckBox.Checked
		End Sub


	End Class

File: DraftingDemo/CSharp/DraftingDemo/TangentsForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WindowsApplication1
{
    public partial class TangentsForm : Form
    {

        public TangentsForm()
        {
            InitializeComponent();
            lineRadioButton.Checked = true;
        }
        
        private void selectButton_Click(object sender, EventArgs e)
        {
            Close();
            DialogResult = DialogResult.OK;

        }

        private void radioButton1_CheckedChanged(object sender, EventArgs e)
        {
            CircleTangents = circlesRadioButton.Checked;
            LineTangents = lineRadioButton.Checked;
            radiusLabel.Enabled = circlesRadioButton.Checked;
            radiusTextBox.Enabled=circlesRadioButton.Checked;
            optionsGruopBox.Enabled=circlesRadioButton.Checked;

        }

        private void circlesRadioButton_CheckedChanged(object sender, EventArgs e)
        {
            CircleTangents = circlesRadioButton.Checked;
            LineTangents = lineRadioButton.Checked;
            radiusLabel.Enabled = circlesRadioButton.Checked;
            radiusTextBox.Enabled=circlesRadioButton.Checked;
            optionsGruopBox.Enabled=circlesRadioButton.Checked;
        }

        private void textBox1_TextChanged(object sender, EventArgs e)
        {
            double val;
            if (Double.TryParse(radiusTextBox.Text, out val))
            {
                TangentRadius = val;
            }

        }
                
        public double TangentRadius = 10;

        public bool LineTangents { get; set; }

        public bool CircleTangents { get; set; }

        public bool TrimTangents { get; set; }

        public bool FlipTangents { get; set; }

        private void checkBox2_CheckedChanged(object sender, EventArgs e)
        {
            TrimTangents = trimCheckBox.Checked;
        }

        private void flipCheckBox_CheckedChanged(object sender, EventArgs e)
        {
            FlipTangents = flipCheckBox.Checked;
        }

        
    }
}
